tests/dist/mpi/test_mpi_functions.cpp

```diff
@@ -45,6 +45,29 @@ TEST_CASE_METHOD(MpiDistTestsFixture, "Test MPI all to all", "[mpi]")
     checkAllocationAndResult(req);
 }
 
+TEST_CASE_METHOD(MpiDistTestsFixture, "Test MPI all to all many times", "[mpi]")
+{
+    int numRuns = 50;
+    int oldNumLocalSlots = nLocalSlots;
+    nLocalSlots = 4;
+    int worldSize = 8;
+    for (int i = 0; i < numRuns; i++) {
+        SPDLOG_INFO("Starting run {}/{}", i + 1, numRuns);
+        // Set up this host's resources
+        setLocalSlots(nLocalSlots, worldSize);
+        auto req = setRequest("alltoall");
+
+        // Call the functions
+        sch.callFunctions(req);
+
+        checkAllocationAndResult(req);
+
+        SLEEP_MS(500);
```

Comment:
Why does this need to sleep? As it's running 50 times and sleeping for half a second each time it's adding 25s of sleeping to the test run.